### PR TITLE
Using Intercom keys with --dart-define

### PR DESCRIPTION
**Pull request by deepak786**
Original PR #320

### CODE REVIEW

File: intercom_flutter/README.md [hunk 1]
```diff
@@ -145,3 +145,63 @@ Add the below script inside body tag in the index.html file located under web fo
 - [ ] handlePush
 - [ ] displayCarousel
 - [ ] displayHelpCenterCollections
+
+## Using of intercom keys with --dart-define
```
Comment (Zazo032) [inline intercom_flutter/README.md]: Wording here should be `Use of intercom...` or `Using intercom...`, and --dart-define should be wrapped with `

File: intercom_flutter/README.md [hunk 2]
```diff
@@ -145,3 +145,63 @@ Add the below script inside body tag in the index.html file located under web fo
...
+
+Use `--dart-define` variables to avoid hardcoding Intercom keys. 
+
+### Pass the Intercom keys with `flutter run` or `flutter build` command using `--dart-define`.
+```dart
+flutter run --dart-define="INTERCOM_APP_ID=appID" --dart-define="INTERCOM_ANDROID_KEY=androidKey" --dart-define="INTERCOM_IOS_KEY=iosKey"
+```
+Note: You can also use `--dart-define-from-file` which is introduced in Flutter 3.7.
+
+### Reading keys in dart side and initialize the SDK.
```
Comment (Zazo032) [inline intercom_flutter/README.md]:
```suggestion
### Reading keys in Dart side and initialize the SDK.
```

File: intercom_flutter/README.md [hunk 3]
```diff
@@ -145,3 +145,63 @@ Add the below script inside body tag in the index.html file located under web fo
...
+```dart
+String appId = String.fromEnvironment("INTERCOM_APP_ID", "");
+String androidKey = String.fromEnvironment("INTERCOM_ANDROID_KEY", "");
+String iOSKey = String.fromEnvironment("INTERCOM_IOS_KEY", "");
+
+Intercom.instance.initialize(appId, iosApiKey: iOSKey, androidApiKey: androidKey);
+```
+
+### Reading keys in android native side and initialize the SDK.
```
Comment (Zazo032) [inline intercom_flutter/README.md]:
```suggestion
### Reading keys in Android native side and initialize the SDK.
```

File: intercom_flutter/README.md [hunk 4]
```diff
@@ -145,3 +145,63 @@ Add the below script inside body tag in the index.html file located under web fo
...
+  dartEnvironmentVariables = project.property('dart-defines')
+      .split(',')
+      .collectEntries { entry ->
+        def pair = new String(entry.decodeBase64(), 'UTF-8').split('=')
+        [(pair.first()): pair.last()]
+      }
+}
+```
+
+* Place dartEnvironmentVariables inside the BuildConfig
```
Comment (Zazo032) [inline intercom_flutter/README.md]:
```suggestion
* Place `dartEnvironmentVariables` inside the build config
```

File: intercom_flutter/README.md [hunk 5]
```diff
@@ -145,3 +145,63 @@ Add the below script inside body tag in the index.html file located under web fo
...
+```
+defaultConfig {
+    ...
+    buildConfigField 'String', 'INTERCOM_APP_ID', "\"${dartEnvironmentVariables.INTERCOM_APP_ID}\""
+    buildConfigField 'String', 'INTERCOM_ANDROID_KEY', "\"${dartEnvironmentVariables.INTERCOM_ANDROID_KEY}\""
+}
+```
+* Read the BuildConfig fields
```
Comment (Zazo032) [inline intercom_flutter/README.md]:
```suggestion

* Read the build config fields
```

**Review by Zazo032 — APPROVED** [review]

**Comment by sirTomasson** [comment]
Heey @Zazo032 ,
I wanted to contribute to this project, and I made a PR just like this, see: https://github.com/v3rm0n/intercom_flutter/pull/320. 
Kinda seems like @deepak786 made this PR using my code 😅.

**Comment by deepak786** [comment]
Hey @sirTomasson 

To your PR, it was not possible to directly add some changes.

And I had to release an update with other changes. I was asking some changes to you but it was taking time to add from you. So I created this PR based on your PR. But the original idea is yours. So thank you.

Don’t worry, I will add your name to the changelog in the next update.

**Pull request merged**